ui/src/main/java/edu/wpi/grip/ui/pipeline/StepController.java

```diff
@@ -82,9 +103,21 @@ private void initialize() {
         new Image(InputStream.class.cast(icon))));
     buttons.getChildren().add(0, exceptionWitnessResponderButtonFactory.create(step, "Step Error"));
 
+    if (step.getInputSockets().stream()
+        .allMatch(inputSocket -> inputSocket.getSocketHint().getView()
+            .equals(SocketHint.View.NONE))) {
```

Comment:
For example:
Desaturate does not allow for user interaction - no arrow is shown.  Filter Contours does -- the arrow is shown.  
![untitled](https://cloud.githubusercontent.com/assets/3799260/17530378/16dc089c-5e2c-11e6-9eeb-1042f875f810.jpg)